tools/circle-upload-codecov.sh

```diff
@@ -2,6 +2,19 @@
 
 set -eo pipefail
 
+if [ -n "${CODECOV_SKIP_UPLOAD:-}" ]; then
+  echo "Skipping Codecov upload (CODECOV_SKIP_UPLOAD is set)"
+  exit 0
+fi
+
+# When CircleCI reruns only failed tests, we end up with partial coverage.
+# The suite-selection logic writes selected suites to files; if they exist and are non-empty,
+# avoid uploading partial coverage.
```

Comment:
This doesn't make sense to me because when a suite failed, the coverage is missing.
If I rerun that suite, and coverage is skipped, the coverage in the PR will be broken.

Could you show me a PR where this was an issue before? AFAIK, it worked as expected, i.e. the partial coverage was added to the already accumulated score.